### PR TITLE
Add EIP-1193 request method

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes https://github.com/trustwallet/trust-web3-provider/issues/83
Fixes https://github.com/trustwallet/trust-web3-provider/issues/70
Changes:

- Code cleanup, remove id buggy filter
- Add `request` method, inherited from `EventEmitter`
- Add `isDebug` flag

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,2 +1,3 @@
 dist/trust-min.js filter=lfs diff=lfs merge=lfs -text
 dist/trust-min.js.map filter=lfs diff=lfs merge=lfs -text
```
Comment (vikmeup) [inline .gitattributes:2]: why do we have different file for iOS and Android?

Comment (hewigovens) [reply .gitattributes:2]: yeah, I also wondered, and it turns out jitpack doesn't support git lfs

Comment (hewigovens) [reply .gitattributes]: android package manager sucks, I will let @madcake  to solve this

**Review by vikmeup — APPROVED** [review]

**Comment by hewigovens** [comment]
@madcake can we remove `lib/src/main/res/raw/trust.js` from git repo and re-use git lfs object ? does this help https://github.com/michel-kraemer/gradle-download-task ?

**Comment by madcake** [comment]
> @madcake can we remove `lib/src/main/res/raw/trust.js` from git repo and re-use git lfs object ? does this help https://github.com/michel-kraemer/gradle-download-task ?

We should continue to use the android library. Just need config Gradle task and use CI.

**Comment by hewigovens** [comment]
Yes, created an issue for this: https://github.com/trustwallet/trust-web3-provider/issues/98, build and publish js directly

**Pull request merged**